src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Symbols/PlaceholderLocalSymbol.vb

```diff
@@ -64,26 +64,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
             compilation As VisualBasicCompilation,
             container As EENamedTypeSymbol,
             syntax As VisualBasicSyntaxNode,
-            isLValue As Boolean) As BoundExpression
+            isLValue As Boolean,
+            diagnostics As DiagnosticBag) As BoundExpression
 
         Friend Shared Function ConvertToLocalType(
             compilation As VisualBasicCompilation,
             expr As BoundExpression,
-            type As TypeSymbol) As BoundExpression
+            type As TypeSymbol,
+            diagnostics As DiagnosticBag) As BoundExpression
 
-            Dim useSiteDiagnostics As HashSet(Of DiagnosticInfo) = Nothing
-            Dim pair = Conversions.ClassifyConversion(expr.Type, type, useSiteDiagnostics)
-            Debug.Assert(pair.Value Is Nothing)
+            Dim syntax = expr.Syntax
+            Dim exprType = expr.Type
 
-            Dim conversionKind = pair.Key
-            Debug.Assert(Conversions.ConversionExists(conversionKind))
-            Debug.Assert((useSiteDiagnostics Is Nothing) OrElse useSiteDiagnostics.All(Function(d) d.Severity < DiagnosticSeverity.Error))
+            Dim conversionKind As ConversionKind
+            If type.IsErrorType() Then
+                diagnostics.Add(type.GetUseSiteErrorInfo(), syntax.GetLocation())
+                conversionKind = Nothing
+            ElseIf exprType.IsErrorType() Then
+                diagnostics.Add(exprType.GetUseSiteErrorInfo(), syntax.GetLocation())
+                conversionKind = Nothing
+            Else
+                Dim useSiteDiagnostics As HashSet(Of DiagnosticInfo) = Nothing
+                Dim pair = Conversions.ClassifyConversion(exprType, type, useSiteDiagnostics)
+
+                diagnostics.Add(syntax, useSiteDiagnostics)
```

Comment:
Does ClassifyConversion ever fail or can we end up with use-site diagnostics here? If not, consider adding back the asserts.